apps/examples/testcase/le_tc/network/tc_net_close.c

```diff
@@ -16,59 +16,97 @@
  *
  ****************************************************************************/
 
-/// @file tc_net_socket.c
-/// @brief Test Case Example for close() API
-#include <tinyara/config.h>
-#include <stdio.h>
-#include <errno.h>
-
-#include <sys/stat.h>
-#include <net/if.h>
-#include <arpa/inet.h>
-#include <netinet/in.h>
-//#include <arch/board/board.h>
-#include <netutils/netlib.h>
-#include <sys/socket.h>
+///file tc_net_socket.c
+///brief Test Case Example for close() API
 
+#include <sys/socket.h>
 #include "tc_internal.h"
 
 /**
-   * @testcase		   :tc_net_close_p
-   * @brief		   :
-   * @scenario		   :
-   * @apicovered	   :close()
-   * @precondition	   :
-   * @postcondition	   :
-   */
+* @testcase          :tc_net_close_p
+* @brief             :close file descriptor
+* @scenario          :open socket and close file descriptor
+* @apicovered        :close()
+* @precondition      :open socket
+* @postcondition     :none
+*/
 static void tc_net_close_p(void)
 {
-
 	int fd = -1;
 	fd = socket(AF_INET, SOCK_STREAM, 0);
+	TC_ASSERT_GEQ("socket", fd, 0);
```

Comment:
If there is a failure on this ASSERT macro, this function will be finished without close(fd). This is a leak.